metaflow/plugins/datatools/s3/s3.py

```diff
@@ -792,6 +792,59 @@ def _head():
 
         return list(starmap(S3Object, _head()))
 
+    def delete(self, key):
+        """
+        Delete a single object in S3.
+
+        Parameters
+        ----------
+        key : str
+            Object to delete. It can be an S3 url or a path suffix.
+        """
+
+        url = self._url(key)
+        src = urlparse(url)
+
+        def _delete(s3, tmp):
+            resp = s3.delete_object(Bucket=src.netloc, Key=src.path.lstrip('/'))
+
+            return resp
+
+        self._one_boto_op(_delete, url)
+
+    def delete_many(self, keys):
```

Comment:
So for this one, I am not sure what the best way forward is:
- either use _one_boto_op and call the `delete_objects` as you do here
- go with the subprocess approach we take for get/put.

I can see pros/cons for both. Let me talk about this internally a little bit.

I haven't reviewed the function in detail yet.

Reply:
Any update on this?